draft-ietf-masque-h3-datagram.md

```diff
@@ -335,238 +229,50 @@ Participant mode:
 : In this mode, the intermediary terminates the data stream and parses all
 Capsule Type and Capsule Length fields it receives.
 
-Each Capsule Type determines whether it is opaque or transparent to
-intermediaries in participant mode: opaque capsules are forwarded unmodified
-while transparent ones can be parsed, added, or removed by intermediaries.
-Intermediaries MAY modify the contents of the Capsule Data field of transparent
-capsule types.
-
-Unless otherwise specified, all Capsule Types are defined as opaque to
-intermediaries. Intermediaries MUST forward all received opaque CAPSULE frames
-in their unmodified entirety. Intermediaries MUST NOT send any opaque CAPSULE
-frames other than the ones it is forwarding. All Capsule Types defined in this
-document are opaque, with the exception of the datagram capsules, see
-{{datagram-capsule}}. Definitions of new Capsule Types MAY specify that the
-newly introduced type is transparent. Intermediaries MUST treat unknown Capsule
-Types as opaque.
-
-Intermediaries respect the order of opaque CAPSULE frames: if an intermediary
-receives two opaque CAPSULE frames in a given order, it MUST forward them in
-the same order.
+Capsules MUST be forwarded unmodified by intermediaries, with the exception of
+the DATAGRAM capsule; the DATAGRAM capsule MAY be parsed, added, or removed by
+intermediaries, see {{datagram-capsule}}. Definitions of new Capsule Types MAY
+specify custom intermediary processing.
```

Comment:
```suggestion
specify optional custom intermediary processing.
```

If you have capsule types that cannot safely be passed verbatim by an intermediary, I don't think you're using The Capsule Protocol.

Reply:
Agreed, added optional.